File: Medium/2433. Find The Original Array of Prefix Xor/2433. Find The Original Array of Prefix Xor.java
class Solution {
    public int[] findArray(int[] pref) {
        int n = pref.length;
        int[] arr = new int[n];
        
        for (int i = 0; i < n; i++) {
            if (i == 0) {
                arr[i] = pref[i];
            } else {
                arr[i] = pref[i] ^ pref[i - 1];
            }
        }
        
        return arr;
    }
}

File: Medium/2433. Find The Original Array of Prefix Xor/2433. Find The Original Array of Prefix Xor.py
from typing import List


class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        n = len(pref)
        arr = [0] * n

        for i in range(n):
            if i == 0:
                arr[i] = pref[i]
            else:
                arr[i] = pref[i] ^ pref[i - 1]

        return arr
